### PR TITLE
Add project preview image API

### PR DESCRIPTION
**Pull request by Nitish145**
### Describe the changes you have made in this PR -
Add `/api/v1/project/:id/image_preview` to GET public project's full image url

### Screenshots of the changes (If any) -
```
{
  "project_preview": "http://localhost:3000/img/default.png"
}
```

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by Nitish145** [comment]
@satu0king @tachyons  please check!!

**Comment by satu0king** [comment]
Looks good to me! Thanks 
@tachyons can you review and merge the same.

File: app/controllers/api/v1/projects_controller.rb [hunk 1]
```diff
@@ -34,6 +34,11 @@ def show
     render json: Api::V1::ProjectSerializer.new(@project, @options)
   end
 
+  def image_preview
+    @project = Project.open.find(params[:id])
```
Comment (tachyons) [inline app/controllers/api/v1/projects_controller.rb:38]: What about private projects? They don't have preview images ?

Comment (Nitish145) [reply app/controllers/api/v1/projects_controller.rb:38]: They do have but @satu0king asked me for public projects only, is that right @satu0king ?

Comment (satu0king) [reply app/controllers/api/v1/projects_controller.rb:38]: Yes idea is that we don't want private project's preview image to get leaked this way. @tachyons is there a better way or its fine to do that?

Comment (tachyons) [reply app/controllers/api/v1/projects_controller.rb:38]: @satu0king We can return 403 if the user is not authorized to view this project, right?

**Pull request merged**